<a href="https://colab.research.google.com/github/dontaedawkins9-ux/IS-362/blob/main/IS362_Week5_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# IS 362 – Week 5 Assignment

This assignment uses Python and pandas along with the nycflights13 dataset to answer the following questions:

1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?

I will use pandas to examine the airport and weather data and research my results to check for possible outliers or incorrect values.

In [1]:
import pandas as pd

# Load the airport and weather datasets
airports_url = "https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv"
weather_url = "https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv"

airports = pd.read_csv(airports_url)
weather = pd.read_csv(weather_url)

# Preview the airport data
airports.head()

,faa,name,lat,lon,alt,tz,dst,tzone
0,04G,Lansdowne Airport,41.130472,-80.619583,1044,-5,A,America/New_York
1,06A,Moton Field Municipal Airport,32.460572,-85.680028,264,-6,A,America/Chicago
2,06C,Schaumburg Regional,41.989341,-88.101243,801,-6,A,America/Chicago
3,06N,Randall Airport,41.431912,-74.391561,523,-5,A,America/New_York
4,09J,Jekyll Island Airport,31.074472,-81.427778,11,-5,A,America/New_York


In [2]:
# Find the five airports with the highest latitude
northernmost = airports.sort_values(by="lat", ascending=False).head(5)

northernmost[["faa", "name", "lat", "lon"]]

,faa,name,lat,lon
417,EEN,Dillant Hopkins Airport,72.270833,42.898333
230,BRW,Wiley Post Will Rogers Mem,71.285446,-156.766003
110,AIN,Wainwright Airport,70.638056,-159.994722
708,K03,Wainwright As,70.613378,-159.860350
152,ATK,Atqasuk Edward Burnell Sr Memorial Airport,70.467300,-157.436000


## Question 1: What is the northernmost airport in the United States?

After sorting the airports by latitude, Dillant Hopkins Airport (EEN) appeared first with a latitude of 72.270833. However, this is an error in the dataset because Dillant Hopkins Airport is located in New Hampshire and is not that far north.

The next airport was Wiley Post-Will Rogers Memorial Airport (BRW) in Utqiagvik, Alaska, with a latitude of 71.285446. I verified this result using FAA information, which places the airport at approximately 71°17' N.

Therefore, based on the dataset and my research, the northernmost airport is **Wiley Post-Will Rogers Memorial Airport (BRW) in Alaska**.

In [3]:
# Find the five airports with the highest longitude
easternmost = airports.sort_values(by="lon", ascending=False).head(5)

easternmost[["faa", "name", "lat", "lon"]]

,faa,name,lat,lon
1290,SYA,Eareckson As,52.712275,174.113620
942,MYF,Montgomery Field,32.475900,117.759000
396,DVT,Deer Valley Municipal Airport,33.411700,112.457000
417,EEN,Dillant Hopkins Airport,72.270833,42.898333
444,EPM,Eastport Municipal Airport,44.910111,-67.012694


## Question 2: What is the easternmost airport in the United States?

I sorted the airport data by longitude and found several values that required additional research. Some airports, including Montgomery Field (MYF) and Deer Valley Municipal Airport (DVT), appeared with positive longitudes even though they are located in the western United States. This indicates errors in the longitude data.

Eareckson Air Station (SYA) in Alaska also appeared because its location is near the 180-degree longitude line, which makes the definition of "easternmost" more complicated.

After reviewing the data and considering the airport locations, I determined that **Eastport Municipal Airport (EPM) in Maine** is the easternmost airport for this analysis. Its longitude in the dataset is approximately -67.012694.

Therefore, my answer is **Eastport Municipal Airport (EPM), Maine**.

In [4]:
# Filter weather data for February 12, 2013
feb12_weather = weather[
    (weather["year"] == 2013) &
    (weather["month"] == 2) &
    (weather["day"] == 12)
]

# Show wind data for the NYC airports
feb12_weather[["origin", "hour", "wind_speed", "wind_gust"]]

,origin,hour,wind_speed,wind_gust
1006,EWR,0,6.90468,NaN
1007,EWR,1,9.20624,NaN
1008,EWR,2,20.71404,25.31716
1009,EWR,3,1048.36058,NaN
1010,EWR,4,12.65858,NaN
...,...,...,...,...
18434,LGA,19,14.96014,21.86482
18435,LGA,20,10.35702,NaN
18436,LGA,21,10.35702,NaN
18437,LGA,22,11.50780,NaN


In [5]:
# Remove unrealistic wind-speed values
valid_weather = feb12_weather[feb12_weather["wind_speed"] < 100]

# Compare the maximum valid wind speed at each airport
valid_weather.groupby("origin")["wind_speed"].max().sort_values(ascending=False)

,wind_speed
origin,
LGA,23.01560
EWR,21.86482
JFK,20.71404


## Question 3: On February 12th, 2013, which New York area airport had the windiest weather?

I filtered the weather data to February 12, 2013 and compared the wind speeds recorded at EWR, JFK, and LGA.

While reviewing the data, I found an unrealistic wind speed of 1048.36058 mph at EWR. I treated this value as an outlier/data error and excluded it from the comparison.

After removing the outlier, the highest wind speeds were:

- LGA: 23.01560 mph
- EWR: 21.86482 mph
- JFK: 20.71404 mph

Therefore, **LaGuardia Airport (LGA)** had the windiest weather on February 12, 2013, with a maximum valid wind speed of approximately **23.02 mph**.